src/api/pagination.ts

```diff
@@ -25,16 +25,58 @@ export function parsePagingQueryInput(val: any) {
   return parsedInput;
 }
 
-interface ParseLimitQueryParams {
-  maxItems: number;
-  errorMsg: string;
+export enum ResourceType {
+  Block,
+  Tx,
+  Event,
+  Burnchain,
+  Contract,
+  Microblock,
+  Token,
 }
 
-export function parseLimitQuery({ maxItems, errorMsg }: ParseLimitQueryParams) {
-  return (val: any) => {
-    const limit = parsePagingQueryInput(val);
-    if (limit > maxItems)
-      throw new InvalidRequestError(errorMsg, InvalidRequestErrorType.invalid_query);
-    return limit;
-  };
+const pagingQueryLimits: Record<ResourceType, { defaultLimit: number; maxLimit: number }> = {
+  [ResourceType.Block]: {
+    defaultLimit: 20,
+    maxLimit: 30,
+  },
+  [ResourceType.Tx]: {
+    defaultLimit: 20,
+    maxLimit: 50,
+  },
+  [ResourceType.Event]: {
+    defaultLimit: 20,
+    maxLimit: 50,
+  },
+  [ResourceType.Burnchain]: {
+    defaultLimit: 96,
+    maxLimit: 250,
+  },
+  [ResourceType.Contract]: {
+    defaultLimit: 20,
+    maxLimit: 50,
+  },
+  [ResourceType.Microblock]: {
+    defaultLimit: 20,
+    maxLimit: 200,
+  },
+  [ResourceType.Token]: {
+    defaultLimit: 50,
+    maxLimit: 200,
+  },
+};
```

Comment:
@CharlieC3 what do you think about these limits? They seem reasonable to me but we could probably make the defaults even smaller to increase performance

Reply:
Considering they're lower than what we currently have and are most familiar with, they look great. I wouldn't want to decrease them drastically in one commit, but I agree it's something we should continue to revisit and adjust.